cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AuthenticationUtils.java

```diff
@@ -85,28 +93,65 @@ public static void validateClientAuthentication(KafkaClientAuthentication authen
             } else if (authentication instanceof KafkaClientAuthenticationPlain) {
                 KafkaClientAuthenticationPlain passwordAuth = (KafkaClientAuthenticationPlain) authentication;
-                volumeList.add(AbstractModel.createSecretVolume(passwordAuth.getPasswordSecret().getSecretName(), passwordAuth.getPasswordSecret().getSecretName(), isOpenShift));
+                createNewSecretVolume(volumeList, passwordAuth.getPasswordSecret().getSecretName(), passwordAuth.getPasswordSecret().getSecretName(), isOpenShift);
             } else if (authentication instanceof KafkaClientAuthenticationScramSha512) {
                 KafkaClientAuthenticationScramSha512 passwordAuth = (KafkaClientAuthenticationScramSha512) authentication;
-                volumeList.add(AbstractModel.createSecretVolume(passwordAuth.getPasswordSecret().getSecretName(), passwordAuth.getPasswordSecret().getSecretName(), isOpenShift));
+                createNewSecretVolume(volumeList, passwordAuth.getPasswordSecret().getSecretName(), passwordAuth.getPasswordSecret().getSecretName(), isOpenShift);
             } else if (authentication instanceof KafkaClientAuthenticationOAuth) {
                 KafkaClientAuthenticationOAuth oauth = (KafkaClientAuthenticationOAuth) authentication;
                 volumeList.addAll(configureOauthCertificateVolumes(oauthVolumeNamePrefix, oauth.getTlsTrustedCertificates(), isOpenShift));
+
+                if (createOAuthSecretVolumes) {
+                    if (oauth.getClientSecret() != null) {
+                        createNewSecretVolume(volumeList, oauth.getClientSecret().getSecretName(), oauth.getClientSecret().getSecretName(), isOpenShift);
+                    }
+                    if (oauth.getAccessToken() != null) {
+                        createNewSecretVolume(volumeList, oauth.getAccessToken().getSecretName(),  oauth.getAccessToken().getSecretName(), isOpenShift);
+                    }
+                    if (oauth.getRefreshToken() != null) {
+                        createNewSecretVolume(volumeList, oauth.getRefreshToken().getSecretName(), oauth.getRefreshToken().getSecretName(), isOpenShift);
+                    }
+                }
             }
         }
     }
 
     /**
+     * Creates a new secret volume and adds it to the list if one with that name does not already exist
+     * 
+     * @param volumeList List where the volume will be added
+     * @param name Volume name
+     * @param path Volume path
+     * @param isOpenShift Indicates whether we run on OpenShift or not
+     */
+    private static void createNewSecretVolume(List<Volume> volumeList, String name, String path, boolean isOpenShift) {
+        // skip if a volume with same Secret name was already added
+        if (!volumeList.stream().anyMatch(vm -> vm.getName().equals(name))) {
```

Comment:
Why is this check necessary? Are surely if there's a dupe that's a programming error and so we should throw?

Reply:
The check already existed in `AuthenticationUtils.configureClientAuthenticationVolumes()` for handling the case where there were multiple TLS trusted certs that needed to be mounted, where it was possible that the same cert secret was used in multiple places in a KafkaBridge or KafkaConnect or KafkaMirrorMaker (1.0) CR.

For MM2 we have a list of clusters, each of which may have TLS certs, password secrets, OAuth tokens, etc and it's possible that the same secret is re-used across different clusters.

Reply:
As an addendum - a single mirror requires a target and source cluster. The target cluster info is used in 2 places - for the Kafka Connect configuration (which ends up in `/tmp/strimzi-connect.properties`), and in the MM2 connectors configuration (as the MM2 connectors do Kafka admin operations on the target topics).

This PR re-uses as much of the connect set-up as possible, which means that the secret volumes and volume mounts for a mirror may have been created by the connect set up, before we try to set up the volumes and volume mounts for the connectors.